flutter/lib/src/sentry_flutter.dart

```diff
@@ -35,27 +35,39 @@ mixin SentryFlutter {
       throw ArgumentError('OptionsConfiguration is required.');
     }
 
-    // It is necessary to initialize Flutter method channels so that
-    // our plugin can call into the native code.
-    WidgetsFlutterBinding.ensureInitialized();
-
     final flutterOptions = SentryFlutterOptions();
+
     // first step is to install the native integration and set default values,
     // so we are able to capture future errors.
-    _addDefaultIntegrations(
-      flutterOptions,
+    final defaultIntegrations = _createDefaultIntegrations(
       isIOSChecker,
       isAndroidChecker,
       packageLoader,
     );
 
+    if (appRunner != null) {
+      final runBefore = () => WidgetsFlutterBinding.ensureInitialized();
+      final runAfter = () => appRunner();
+      final rzgi = RunZoneGuardedIntegration(runBefore, runAfter);
+      flutterOptions.addIntegration(rzgi);
+
+      for (final defaultIntegration in defaultIntegrations) {
+        flutterOptions.addIntegration(defaultIntegration);
+      }
+    } else {
+      WidgetsFlutterBinding.ensureInitialized();
+      for (final defaultIntegration in defaultIntegrations) {
+        flutterOptions.addIntegration(defaultIntegration);
+      }
+    }
+
     await _initDefaultValues(flutterOptions);
 
     await Sentry.init(
       (options) async {
         await optionsConfiguration(options);
       },
-      appRunner: appRunner,
+      appRunner: null, // Handled by RunZoneGuardedIntegration
```

Comment:
not sure if I follow, how one would pass the `appRunner` instance?

Reply:
so the problem with this is, a Flutter app may be used by the `sentry` package too and not only `senttry_flutter`, this would not be possible anymore, how would you pass the `appRunner` instance? if thru `RunZoneGuardedIntegration`, how one creates its own `runZonedGuarded`?

I think the creation of the `RunZoneGuardedIntegration` should still be under the pure dart package, and get the `runBefore`, `runAfter` via params, so it'd work for both packages seamlessly.

Reply:
`runBefore` would be always `NoOp` probably cus the dart package has no Flutter deps so `ensureInitialized` does not exist.

Reply:
- Ah ok, i assumed (incorrectley) we don't need the `runZoneGuarded` when only using the pure dart `sentry` package.
- Also, one of my assumtions was to only run the other ingeration inside the `runZoneGuarded` if we provide the `appRunner`, this is stil the intended behaviour?

Reply:
My assumption was that when using only using the pure `sentry` dart package, users would just provide their `appRunnter` callback themselves, as there would not be a wrapping `sentry_flutter` package. In the `sentry_flutter` package we handle it in the RunZoneGuarded integration, therefore we don't need to pass it, as it has it's own `RunZoneGuardedIntegration`. Are these assumptions wrong?

Reply:
Also, the `RunZoneGuarded` integration has the responsibility to run all other integration, also the ones created inside the dart only `flutter` package. but overall, i'm not too happy with a integration running other integrations within, as we have to keep tabs on which ones were run this way with the boolean flag introduced.

Reply:
I'm a bit confused right now but I guess this code snippet will help, see the use cases, both should be possible.

```
Future<void> main() async {
  // 1st: pure dart package, running a Flutter app
    await Sentry.init(
      (options) => options
        ..dsn = dsn
      appRunner: runApp, // runApp is available, it does make sense that we automatically execute a runZonedGuarded integration, so events are automatically caught by default
    );

  // 2nd: pure dart package, running a Flutter app or console, does not matter
    await Sentry.init(
      (options) => options
        ..dsn = dsn 
        // appRunner is not available, it does not make make sense that we automatically execute a runZonedGuarded integration, as we don't know in which context
        // its App is run, so they need to create their own runZonedGuarded
    );

  // the solution for the 2nd case would be as easy as:
  // init sentry without appRunner
    await Sentry.init(
      (options) => options
        ..dsn = dsn 
    );
  // create your own runZonedGuarded
    runZonedGuarded(() async {
      // run your own WidgetsFlutterBinding.ensureInitialized();
      await runApp();
    }, (exception, stackTrace) async {
      // capture yourself
      await hub.captureException(exception, stackTrace: stackTrace);
    });
}

Future<void> runApp() async {
  // run its own App
}
```

Reply:
unresolving so that I can copy paste the comment URL otherwise it does not work